src/common/task.h

```diff
@@ -13,6 +13,40 @@
 
 typedef uint8_t TaskSpec;
 
+class TaskExecutionSpec {
+ public:
+  TaskExecutionSpec(const std::vector<ObjectID> execution_dependencies,
+                    TaskSpec *spec,
+                    int64_t task_spec_size);
+  TaskExecutionSpec(TaskExecutionSpec *execution_spec);
+
+  std::vector<ObjectID> ExecutionDependencies();
+  int64_t SpecSize();
+  TaskSpec *Spec();
+  int64_t NumDependencies();
+  int DependencyIdCount(int64_t dependency_index);
+  ObjectID DependencyId(int64_t dependency_index, int64_t id_index);
+  /**
```

Comment:
not important, but for new things, we might as well start using the `///` style instead of `/** */`